features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java

```diff
@@ -0,0 +1,156 @@
+
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {
+        "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
+        "classpath:/META-INF/opennms/applicationContext-soa.xml",
+        "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-databasePopulator.xml"
+})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase(reuseDatabase = false, tempDbClass = MockDatabase.class)
+@Transactional
+public class NMS15448_IT implements InitializingBean {
+    @Autowired
+    DatabasePopulator databasePopulator;
+
+    @Autowired
+    TransactionTemplate transactionOperations;
+
+    private OnmsDistPoller distPoller;
+
+    @Override
+    public void afterPropertiesSet() {
+        BeanUtils.assertAutowiring(this);
+    }
+
+    @Before
+    public void setUp() {
+        databasePopulator.populateDatabase();
+        distPoller = databasePopulator.getDistPollerDao().whoami();
+    }
+
+    private OnmsAlarm buildAlarm(final OnmsEvent event, boolean acknowledged) {
+        final OnmsAlarm alarm = new OnmsAlarm();
+        alarm.setDistPoller(distPoller);
+        alarm.setUei(event.getEventUei());
+        alarm.setAlarmType(OnmsAlarm.PROBLEM_TYPE);
+        alarm.setNode(databasePopulator.getNode1());
+        alarm.setDescription("This is a test alarm");
+        alarm.setLogMsg("this is a test alarm log message");
+        alarm.setCounter(1);
+        alarm.setIpAddr(InetAddressUtils.getInetAddress("192.168.1.1"));
+        alarm.setSeverity(OnmsSeverity.NORMAL);
+        alarm.setFirstEventTime(event.getEventTime());
+        alarm.setLastEvent(event);
+        alarm.setServiceType(databasePopulator.getServiceTypeDao().findByName("ICMP"));
+        if (acknowledged) {
+            alarm.setAlarmAckUser("foobar");
+            alarm.setAlarmAckTime(new Date());
+        }
+        return alarm;
+    }
+
+    @Test
+    public void testSomething() {
+        final DefaultSurveillanceViewService surveillanceViewService = new DefaultSurveillanceViewService();
+        surveillanceViewService.setTransactionOperations(transactionOperations);
+        surveillanceViewService.setMonitoredServiceDao(databasePopulator.getMonitoredServiceDao());
+        surveillanceViewService.setOutageDao(databasePopulator.getOutageDao());
+        surveillanceViewService.setAlarmDao(databasePopulator.getAlarmDao());
+        final OnmsCategory cat1 = databasePopulator.getCategoryDao().findByName("Routers");
+        final OnmsCategory cat2 = databasePopulator.getCategoryDao().findByName("DEV_AC");
+
+        final List<OnmsAlarm> alarmsBefore = surveillanceViewService.getAlarmsForCategories(Sets.newHashSet(cat1), Sets.newHashSet(cat2));
+
+        Assert.assertEquals(1, alarmsBefore.size());
+
+        final OnmsEvent event1 = databasePopulator.buildEvent(distPoller);
+        event1.setEventCreateTime(new Date(1436881548292L));
```

Comment:
Maybe make the `Date` a constant, so it's clear it's just the same for every item.

Reply:
done.